src/java.desktop/macosx/classes/sun/lwawt/macosx/CPlatformWindow.java

```diff
@@ -965,6 +966,33 @@ public boolean isFullScreenMode() {
         return isFullScreenMode;
     }
 
+    private void waitForWindowState(int state) {
+        if (peer.getState() == state) {
+            return;
+        }
+
+        Object lock = new Object();
+        WindowStateListener wsl = new WindowStateListener() {
+            public void windowStateChanged(WindowEvent e) {
+                synchronized (lock) {
+                    if (e.getNewState() == state) {
+                        lock.notifyAll();
+                    }
+                }
+            }
+        };
+
+        target.addWindowStateListener(wsl);
+        if (peer.getState() != state) {
+            synchronized (lock) {
+                try {
+                    lock.wait();
```

Comment:
Are we sure that waiting forever is a good thing? Can this request be ignored by the OS for any reason?

Reply:
It is very unlikely that we are waiting forever here unless there are some serious problems with corresponding platform API. 
I understand your concern and agree that waiting for some reasonable time instead of forever is more suitable here.
@alisenchung Can you open a bug to replace forever waiting with timeout.